Update docs for default 2 GB Kibana instances (#6121)
elastic/docs-content: deploy-manage/upgrade/deployment-or-cluster/upgrade-on-ech.md

Refer to [Plan for production](/deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning.md) for more information about High Availability (HA) on ECH.

## Check your {{kib}} instance size 

:::{include} /deploy-manage/upgrade/deployment-or-cluster/_snippets/kib-instance-size.md
:::

[Learn how to edit settings for existing deployments](/deploy-manage/deploy/elastic-cloud/configure.md).

:::{note}

> {{kib}} instances in new {{ech}} deployments have 2 GB of RAM by default. 

Reviewer: ```suggestion
{{kib}} instances in {{ech}} deployments created in May 2026 or later have 2 GB of RAM by default. 
```

Refer to [Plan for production](/deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning.md) for more information about High Availability (HA) on ECH.

> ## Check your {{kib}} instance size 

Reviewer: we might want to put this on all of the upgrade pages

Reviewer: yes according to the proposal doc (internal) this applies to on-prem + self too

Refer to [Plan for production](/deploy-manage/deploy/elastic-cloud/elastic-cloud-hosted-planning.md) for more information about High Availability (HA) on ECH.

> ## Increase your {{kib}} instance size 

Reviewer 2: We're not sure where do users start from and should make this future proof so I'll leave a few suggestions for this section
```suggestion
## Check your {{kib}} instance size 
```

> {{kib}} needs more memory to support growing built-in functionality. 

Reviewer 2: ```suggestion
{{kib}} needs enough memory to support growing built-in functionality.
```

> Before upgrading your environment, we recommend increasing the size of your {{kib}} instances to at least 2 GB of RAM per availability zone. Learn how to edit settings for existing deployments in [Configure](/deploy-manage/deploy/elastic-cloud/configure.md).

Reviewer 2: ```suggestion
Before upgrading your environment to {{stack}} version 9 or higher, we recommend increasing the size of your {{kib}} instances to at least 2 GB of RAM per availability zone. Learn how to edit settings for existing deployments in [Configure](/deploy-manage/deploy/elastic-cloud/configure.md).
```
Not all users are upgrading to the latest, so we may want to scope this to 9 and above? It's not strictly 9.4, but it's also maybe not super needed for versions < 9.x. Also fine if we don't do that.

> Before upgrading your environment, we recommend increasing the size of your {{kib}} instances to at least 2 GB of RAM per availability zone. Learn how to edit settings for existing deployments in [Configure](/deploy-manage/deploy/elastic-cloud/configure.md).

Reviewer 2: There or in the first paragraph, we might want to add that this is rather specific to Platinum and Enterprise users as the added features at the origin of this are at those levels of subscription. This helps users on lower tiers not have to worry too much about this, since they actually may also be users that would be more concerned about the cost implications of this recommendation.

Author: Good point! Updating.

Increasing the memory of your instances ensures optimal performance of new and existing Platinum and Enterprise features. With less than 2 GB of RAM, you can face service interruptions when using features such as Security detection rules, reporting, workflows, and Agent Builder.

> {applies_to}`stack: ga 9.4+` {{kib}} instances in new {{ech}} deployments have 2 GB of RAM by default. 

Reviewer 2: ```suggestion
{{kib}} instances in new {{ech}} deployments have 2 GB of RAM by default. 
```
Same comment as before + could maybe be a note?
